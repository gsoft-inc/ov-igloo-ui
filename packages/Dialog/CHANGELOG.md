# @igloo-ui/dialog

## 0.8.4

### Patch Changes

- 8bf2e79: Version bump to acknowledge button patch version
- Updated dependencies [8bf2e79]
  - @igloo-ui/modal@1.6.5

## 0.8.3

### Patch Changes

- 6b00f08: Updated dependencies
  - @igloo-ui/modal@1.6.4

## 0.8.2

### Patch Changes

- 33ecc10: Updated the width behavior of the buttons in Dialog.
- Updated dependencies [40eb117]
  - @igloo-ui/modal@1.6.2

## 0.8.1

### Patch Changes

- 37822b2: Fixed the style of the modal and Dialog headers and the spacing of the close button.
- Updated dependencies [37822b2]
  - @igloo-ui/modal@1.6.1

## 0.8.0

### Minor Changes

- 4846c59: Wrapped Igloo CSS in @layer to isolate it.

### Patch Changes

- Updated dependencies [4846c59]
  - @igloo-ui/button@0.8.0
  - @igloo-ui/modal@1.6.0

## 0.7.4

### Patch Changes

- cfb5619: Updated hopper token versions and button versions.
- Updated dependencies [cfb5619]
- Updated dependencies [cfb5619]
  - @igloo-ui/button@0.7.7
  - @igloo-ui/modal@1.5.4

## 0.7.3

### Patch Changes

- 26614d0: Fixed the spacing for titles and buttons inside the Modal and Dialog components to match figma.
- Updated dependencies [26614d0]
  - @igloo-ui/modal@1.5.3

## 0.7.2

### Patch Changes

- 7d36815: Updated hopper token versions and igloo icon versions.
- Updated dependencies [7d36815]
  - @igloo-ui/button@0.7.4
  - @igloo-ui/modal@1.5.2

## 0.7.1

### Patch Changes

- 0dda883: dismissOnEscape is now optional. 'contain' was removed from FocusScope inside Modal. This was causing errors in OV when dialog displayed above modal.
- Updated dependencies [0dda883]
  - @igloo-ui/modal@1.5.1

## 0.7.0

### Minor Changes

- 636004c: Added dismissOnEscape to Modal and Dialog. Also fixed the issue where the modal didn't gain focus after opening.

### Patch Changes

- Updated dependencies [636004c]
  - @igloo-ui/modal@1.5.0

## 0.6.2

### Patch Changes

- a2df55d: Updated to latest Hopper Token dependency
- Updated dependencies [a2df55d]
  - @igloo-ui/button@0.7.3
  - @igloo-ui/modal@1.4.3

## 0.6.1

### Patch Changes

- 9739155: Gave ColorPicker the new Wokleap look. Color now uses css variables to update its color. Fixed spacing issues in tooltip. Updated hopper token version.
- Updated dependencies [9739155]
  - @igloo-ui/button@0.7.1
  - @igloo-ui/modal@1.4.2

## 0.6.0

### Minor Changes

- 8eeac98: Rebrand of dialog and modal / media query fix in rems

### Patch Changes

- Updated dependencies [8eeac98]
  - @igloo-ui/button@0.7.0
  - @igloo-ui/modal@1.4.0

## 0.5.1

### Patch Changes

- cfdea66: Fixed Dialog's button width so that they are equal. Also fixed dialog header's styling. It was using a no longer used class.

## 0.5.0

### Minor Changes

- 522d3cf: Added a loading feature to the Dialog when the validate button is clicked.

## 0.4.1

### Patch Changes

- bf4ddef: Updated storybook version and linting. Made linting fixes in these components.
- Updated dependencies [bf4ddef]
  - @igloo-ui/button@0.5.2
  - @igloo-ui/modal@1.3.3

## 0.4.0

### Minor Changes

- c7273b3: Update Dialog prop, subTitle, to be a ReactNode.

## 0.3.2

### Patch Changes

- 0e3cc82: added troubleshooting section in the readme for workaround about Closing Dialog on top of Modal will close both
- Updated dependencies [0e3cc82]
  - @igloo-ui/modal@1.2.3

## 0.3.1

### Patch Changes

- a41e1d4: Fix: moved destructuring of props to function parameters so that storybook will automatically show the default value in the props table. It was done in Modal because it needed to send props to useDialog. In this case, @default annotation was used in comments.
- Updated dependencies [a41e1d4]
  - @igloo-ui/button@0.4.1
  - @igloo-ui/modal@1.2.2

## 0.3.0

### Minor Changes

- bf0342e: Refactor: Replaced react spring and react popper with framer motion and/or floating-ui

### Patch Changes

- Updated dependencies [bf0342e]
  - @igloo-ui/modal@1.2.0

## 0.2.0

### Minor Changes

- 386e7cd: Reduced the padding for the modal and dialog components. Also added steps and buttons to the modal. The carousel just had a minor style update on the secondary button.

### Patch Changes

- Updated dependencies [386e7cd]
  - @igloo-ui/modal@1.1.0

## 0.1.4

### Patch Changes

- d1ee42f: Fix: Added an if statement on the dialog demo page so that the dialog code doesn't stick around. This will need to be updated once react-spring fixes the javascript error.

## 0.1.3

### Patch Changes

- 6ee9c45: Updated anything causing errors or warnings in the browser console
- Updated dependencies [6ee9c45]
  - @igloo-ui/button@0.3.7
  - @igloo-ui/modal@1.0.3

## 0.1.2

### Patch Changes

- a1a4642: Add danger props for render a desctructive actions

## 0.1.1

### Patch Changes

- 0640409: udpate classnames to 2.3.2
- Updated dependencies [0640409]
- Updated dependencies [0640409]
- Updated dependencies [0640409]
  - @igloo-ui/button@0.3.6
  - @igloo-ui/modal@1.0.2

## 0.1.0

### Minor Changes

- b734729: Initial release of the dialog component

### Patch Changes

- Updated dependencies [b734729]
  - @igloo-ui/modal@1.0.0
