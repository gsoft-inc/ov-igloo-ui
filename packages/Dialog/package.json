{
  "name": "@igloo-ui/dialog",
  "version": "0.9.2",
  "main": "dist/Dialog.js",
  "module": "dist/Dialog.js",
  "types": "dist/Dialog.d.ts",
  "styles": "dist/dialog.css",
  "publishConfig": {
    "access": "public"
  },
  "repository": "git+https://github.com/gsoft-inc/ov-igloo-ui.git",
  "author": "Officevibe Inc.",
  "contributors": [
    "Vicky Comeau vicky.comeau@gsoft.com"
  ],
  "license": "Apache-2.0",
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "rollup -c rollup.config.js"
  },
  "dependencies": {
    "@igloo-ui/tokens": "^2.1.0",
    "classnames": "^2.3.2",
    "@igloo-ui/modal": "^1.8.1",
    "@igloo-ui/button": "^0.9.1",
    "@hopper-ui/tokens": "^4.0.0"
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "IE >= 11"
  ]
}
