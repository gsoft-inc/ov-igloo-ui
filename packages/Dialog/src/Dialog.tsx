import * as React from 'react';

import Button from '@igloo-ui/button';
import Modal from '@igloo-ui/modal';
import cx from 'classnames';

import './dialog.scss';

export interface DialogProps extends React.ComponentProps<'div'> {
  /** Add a specific class to the dialog */
  className?: string;
  /** Add a data-test tag for automated tests */
  dataTest?: string;
  /** The text for the dismiss button */
  dismissText?: string;
  /** Whether the dialog is open or not */
  isOpen: boolean;
  /** Handler that is called when the overlay should close */
  onDismiss: () => void;
  /** Handler that is called when the dialog
   * is closed and the animation is complete */
  onAfterDismiss?: () => void;
  /** Handler that is called when the validate button is clicked */
  onValidate: () => void;
  /** The text for the subtitle of the dialog */
  subTitle?: string;
  /** The text for the title of the dialog */
  title: string;
  /** The text for the validate button */
  validateText: string;
  /** A specific variant of dialog used for destructive actions */
  danger?: boolean;
}

const Dialog: React.FunctionComponent<DialogProps> = ({
  className,
  dataTest,
  dismissText,
  isOpen,
  onDismiss,
  onAfterDismiss,
  onValidate,
  subTitle,
  title,
  validateText,
  danger,
}: DialogProps) => {
  const classes = cx('ids-dialog', className);

  return (
    <Modal
      className={classes}
      dataTest={dataTest}
      isClosable
      isOpen={isOpen}
      size="small"
      onClose={onDismiss}
      onAfterClose={onAfterDismiss}
      title={title}
    >
      {subTitle && <div className="ids-dialog__sub-title">{subTitle}</div>}
      <div className="ids-dialog__buttons">
        {dismissText && onDismiss && (
          <Button
            appearance="secondary"
            onClick={onDismiss}
            className="ids-dialog__dismiss-btn"
          >
            {dismissText}
          </Button>
        )}
        <Button
          appearance={danger ? 'danger' : 'primary'}
          onClick={onValidate}
          className="ids-dialog__validate-btn"
        >
          {validateText}
        </Button>
      </div>
    </Modal>
  );
};

export default Dialog;
