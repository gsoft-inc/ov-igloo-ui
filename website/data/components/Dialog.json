[
  [
    {
      "tags": {},
      "filePath": "/Users/franck.gaudin/Devel/Igloo/components/packages/Dialog/src/Dialog.tsx",
      "description": "",
      "displayName": "Dialog",
      "methods": [],
      "props": {
        "className": {
          "defaultValue": null,
          "description": "Add a specific class to the dialog",
          "name": "className",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "dataTest": {
          "defaultValue": null,
          "description": "Add a data-test tag for automated tests",
          "name": "dataTest",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "dismissText": {
          "defaultValue": null,
          "description": "The text for the dissmiss button",
          "name": "dismissText",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "isOpen": {
          "defaultValue": null,
          "description": "Whether the dialog is open or not",
          "name": "isOpen",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": true,
          "type": { "name": "boolean" }
        },
        "onDismiss": {
          "defaultValue": null,
          "description": "Handler that is called when the overlay should close",
          "name": "onDismiss",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": true,
          "type": { "name": "() => void" }
        },
        "onAfterDismiss": {
          "defaultValue": null,
          "description": "Handler that is called when the dialog\nis closed and the animation is complete",
          "name": "onAfterDismiss",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": false,
          "type": { "name": "() => void" }
        },
        "onValidate": {
          "defaultValue": null,
          "description": "Handler that is called when the validate button is clicked",
          "name": "onValidate",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": true,
          "type": { "name": "() => void" }
        },
        "subTitle": {
          "defaultValue": null,
          "description": "The text for the sub title of the dialog",
          "name": "subTitle",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": false,
          "type": { "name": "string" }
        },
        "title": {
          "defaultValue": null,
          "description": "The text for the title of the dialog",
          "name": "title",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": true,
          "type": { "name": "string" }
        },
        "validateText": {
          "defaultValue": null,
          "description": "The text for the validate button",
          "name": "validateText",
          "parent": {
            "fileName": "components/packages/Dialog/src/Dialog.tsx",
            "name": "DialogProps"
          },
          "declarations": [
            {
              "fileName": "components/packages/Dialog/src/Dialog.tsx",
              "name": "DialogProps"
            }
          ],
          "required": true,
          "type": { "name": "string" }
        }
      }
    }
  ]
]
