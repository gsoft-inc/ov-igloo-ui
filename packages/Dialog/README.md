# Dialog

The Dialog is much like the Modal component, except it is simpler. It contains a title, sub title, dismiss action and a validate action.

<Example is="custom" />

<ReferenceLinks is="custom" />

## Installation

To install `@igloo-ui/dialog` in your project, you will need to run the following command using [npm](https://www.npmjs.com/):

```bash
npm install @igloo-ui/dialog
```

If you prefer [Yarn](https://classic.yarnpkg.com/en/), use the following command instead:

```bash
yarn add @igloo-ui/dialog
```

## Usage

Then to use the component in your code just import it!

```jsx
import Dialog from '@igloo-ui/dialog';
import Button from '@igloo-ui/button';

const [open, setOpen] = React.useState(false);
const handleValidate = () => {
    alert("You said yes");
    setOpen(false);
};

<Button appearance="secondary" onClick={() => setOpen(true)}>
    open
</Button>
<Dialog
    title="Dialog title"
    subTitle="This is a sub title"
    dismissText="Cancel"
    validateText="Confirm"
    isOpen={open}
    onDismiss={() => setOpen(false)}
    onValidate={handleValidate}
/>
```

## Troubleshooting

**When a Dialog opens above a Modal**

In this case when you close the Dialog the Modal closes automatically, here is a workaround to remedy this.

To solve the problem, set the `isClosable` property of the Modal to `false` when the Dialog component is up and then return to `true` when it closes.
