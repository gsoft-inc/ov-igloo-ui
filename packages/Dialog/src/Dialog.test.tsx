/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

import Dialog from './Dialog';

describe('Dialog', () => {
  test('It should render without error and a snapshot', () => {
    const isOpen = true;
    const { asFragment } = render(
      <Dialog
        dataTest="dialog1"
        title="Dialog title"
        subTitle="Dialog sub title"
        dismissText="No"
        validateText="Yes"
        onDismiss={() => !isOpen}
        onValidate={() => console.log('validated')}
        isOpen={isOpen}
      />
    );
    expect(screen.getByTestId('dialog1')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  test('It should render the title', () => {
    const isOpen = true;
    render(
      <Dialog
        title="Dialog title"
        subTitle="Dialog sub title"
        dismissText="No"
        validateText="Yes"
        onDismiss={() => !isOpen}
        onValidate={() => console.log('validated')}
        isOpen={isOpen}
      />
    );

    const title = screen.getByText('Dialog title');
    expect(title).toBeInTheDocument();
  });

  test('It should render the sub title', () => {
    const isOpen = true;
    render(
      <Dialog
        title="Dialog title"
        subTitle="Dialog sub title"
        dismissText="No"
        validateText="Yes"
        onDismiss={() => !isOpen}
        onValidate={() => console.log('validated')}
        isOpen={isOpen}
      />
    );

    const subtitle = screen.getByText('Dialog sub title');
    expect(subtitle).toBeInTheDocument();
  });

  test('It should render the dissmiss button', () => {
    const isOpen = true;
    render(
      <Dialog
        title="Dialog title"
        subTitle="Dialog sub title"
        dismissText="No"
        validateText="Yes"
        onDismiss={() => !isOpen}
        onValidate={() => console.log('validated')}
        isOpen={isOpen}
      />
    );

    const dismissButton = screen.getByText('No');
    expect(dismissButton).toBeInTheDocument();
  });

  test('It should render the validate button', () => {
    const isOpen = true;
    render(
      <Dialog
        title="Dialog title"
        subTitle="Dialog sub title"
        dismissText="No"
        validateText="Yes"
        onDismiss={() => !isOpen}
        onValidate={() => console.log('validated')}
        isOpen={isOpen}
      />
    );

    const validateButton = screen.getByText('Yes');
    expect(validateButton).toBeInTheDocument();
  });
});
