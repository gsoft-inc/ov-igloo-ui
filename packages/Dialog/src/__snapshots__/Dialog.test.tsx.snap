// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog It should render without error and a snapshot 1`] = `<DocumentFragment />`;
